### PR TITLE
nmstate-autoconf: introduce nmstate-autoconf support

### PR DESCRIPTION
**Pull request by ffmancera**
Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

### CODE REVIEW

**Comment by ffmancera** [comment]
@cathay4t This is now ready for review. Thank you!

File: doc/nmstate-autoconf.8.in [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+.B *\fR matches everything
+.br
+.B ?\fR matches any single character
+.br
+.B [seq]\fR matches any character in seq
+.br
+.B [!seq]\fR matches any character not in seq
+.RE
+.PP
+For example, to show all interfaces starts with eth:
```
Comment (cathay4t) [inline doc/nmstate-autoconf.8.in]: not `show all interfaces`, but `enable LLDP auto configuration on all interfaces`.

File: doc/nmstate-autoconf.8.in [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+.RS
+nmstate-autoconf eth\\*
+.br
+# The backslash is required to stop shell expanding '*' to file names.
+.RE
+.SH OPTIONS
+.B --bond-vlans
```
Comment (cathay4t) [inline doc/nmstate-autoconf.8.in]: If this is enabled by default, basically you don't have any way to disable it with current code.

**Review by cathay4t — CHANGES_REQUESTED** [review]

**Comment by cathay4t** [comment]
Having a `--dry-run` by print out the desired state could help debugging and playing.

File: doc/nmstate-autoconf.8.in (hunk 2 above)
Comment (ffmancera) [reply doc/nmstate-autoconf.8.in]: Yes exactly, this what we discussed. The idea was to extend this in the future and introduce different patterns.

**Comment by ffmancera** [comment]
> Having a `--dry-run` by print out the desired state could help debugging and playing.

Done, thank you!

File: doc/nmstate-autoconf.8.in (hunk 2 above)
Comment (cathay4t) [reply doc/nmstate-autoconf.8.in]: Then this option is useless. Please remove it and add it back when required.

**Review by cathay4t — CHANGES_REQUESTED** [review]

**Review by cathay4t — APPROVED** [review]

**Pull request merged**